pkg/skaffold/kubernetes/context/context.go

```diff
@@ -50,12 +50,7 @@ func UseKubeContext(overrideKubeContext string) {
 // If UseKubeContext was called before, the CurrentContext will be overridden.
 // The result will be cached after the first call.
 func GetRestClientConfig() (*restclient.Config, error) {
-	rawConfig, err := getRawKubeConfig()
-	if err != nil {
-		return nil, err
-	}
-	clientConfig := clientcmd.NewNonInteractiveClientConfig(rawConfig, kubeContext, &clientcmd.ConfigOverrides{CurrentContext: kubeContext}, nil)
```

Comment:
The reason for the detour to the raw kubeconfig was that the `DeferredLoadingClientConfig` caches the (REST)`clientConfig` with the current kubecontext (see vendor/k8s.io/client-go/tools/clientcmd/merged_client_builder.go:62). This means that the RESTClientConfig used by Skaffold will always work with the original kube-context, regardless of whether that was changed by Skaffold. Thus, the k8s REST calls would work with a different context than direct `kubectl` invocations (via `CurrentConfig` pkg/skaffold/kubernetes/context/context.go:59). This is even confusing to explain -- did I make it understandable?

So in essence, without the detour via the raw kube-config, there can be a misalignment of REST and `kubectl` calls which we should avoid.

---

Also see new test-case "change context after first execution"